### PR TITLE
Renaming getDriver to getAppium driver

### PR DESCRIPTION
**Pull request by Jared1048**
This is to prevent tests from having getDriver().getDriver() to access the core Appium methods.

### CODE REVIEW

File: src/main/java/com/joss/conductor/mobile/ConductorConfig.java [hunk 1]
```diff
@@ -443,4 +448,18 @@ private void putCustomCapabilities(Object keyCustomCapabilities) {
     private void putCustomCapabilities(Map<String, Object> customCapabilities) {
         this.customCapabilities.putAll(customCapabilities);
     }
+
+    public SauceOnDemandAuthentication getAuthentication() {
```
Comment (jossjacobo) [inline src/main/java/com/joss/conductor/mobile/ConductorConfig.java]: We should rename this to `getSauceAuthentication()` or something along those lines, it's not super obvious what `getAuthentication()` is for.

File: src/main/java/com/joss/conductor/mobile/ConductorConfig.java [hunk 2]
```diff
@@ -443,4 +448,18 @@ private void putCustomCapabilities(Object keyCustomCapabilities) {
...
+        String sauceUserName = System.getProperty("SAUCE_USERNAME");
+        String sauceAccessKey = System.getProperty("SAUCE_ACCESS_KEY");
```
Comment (jossjacobo) [inline src/main/java/com/joss/conductor/mobile/ConductorConfig.java]: We should make these actual config fields that people can fill out and not just rely on the environment variable.

...Things I should've done yesterday but didn't haha

File: src/main/java/com/joss/conductor/mobile/Locomotive.java [hunk 3]
```diff
@@ -75,6 +80,14 @@ public Locomotive(ConductorConfig configuration, AppiumDriver driver) {
         init(configuration, driver);
     }
 
+    public AppiumDriver getAppiumDriver() {
+        return driver.get();
+    }
+
+    public void setDriver(AppiumDriver d) {
```
Comment (jossjacobo) [inline src/main/java/com/joss/conductor/mobile/Locomotive.java]: Should also rename this method sig to `setAppiumDriver(Appiumdriver driver)`.

File: src/main/java/com/joss/conductor/mobile/ConductorConfig.java (hunk 1 above)
Comment (Jared1048) [reply src/main/java/com/joss/conductor/mobile/ConductorConfig.java]: good point

File: src/main/java/com/joss/conductor/mobile/ConductorConfig.java (hunk 2 above)
Comment (Jared1048) [reply src/main/java/com/joss/conductor/mobile/ConductorConfig.java]: done. think I fixed it proper but see next commit

**Comment by konwinkler** [comment]
One general note for anyone using this:

File: src/main/java/com/joss/conductor/mobile/ConductorConfig.java [hunk 4]
```diff
@@ -443,4 +458,16 @@ private void putCustomCapabilities(Object keyCustomCapabilities) {
     private void putCustomCapabilities(Map<String, Object> customCapabilities) {
         this.customCapabilities.putAll(customCapabilities);
     }
+
+    public SauceOnDemandAuthentication getSauceAuthentication(String sauceUserName, String sauceAccessKey) {
+
+        if (sauceUserName == null) {
+            throw new InvalidArgumentException("Env var for SAUCE_USERNAME cannot be null");
```
Comment (jossjacobo) [inline src/main/java/com/joss/conductor/mobile/ConductorConfig.java]: Update the error message to use the `config.yaml` and for they key `sauceUserName` and `sauceAccessKey` fields.

**Pull request merged**